File: frontend/src/types/subscription.ts
import { Tier } from "./tier";

export interface UserSubscription {
  id: number;
  user_id: string;
  tier_id: number;
  start_date: string;
  end_date: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  tier: Tier;
}

export interface TierSubscription {
  tier_id: number;
  status: string;
  expires_at: string;
}

export interface CheckoutSessionResponse {
  checkout_session_id: string;
  url: string;
}

File: app/schemas/subscription.py
from pydantic import BaseModel
from datetime import datetime
from .tier import TierWithProducts

class UserSubscriptionBase(BaseModel):
    tier_id: int
    start_date: datetime
    end_date: datetime
    is_active: bool = True

class UserSubscriptionCreate(UserSubscriptionBase):
    user_id: str

class UserSubscriptionResponse(UserSubscriptionBase):
    id: int
    user_id: str
    created_at: datetime
    updated_at: datetime | None
    tier: TierWithProducts
    stripe_subscription_id: str | None

    class Config:
        from_attributes = True
